proto/src/logcollector.proto

```diff
@@ -0,0 +1,52 @@
+syntax = "proto3";
+
+package org.eln2.proto.logcollector;
+import "types.proto";
+
+// Everything in this file is still in flux. It will be locked down prior to first alpha release.
+service LogService {
+	rpc Log(LogEntry) returns (LogResponse);
+}
+
+message LogEntry {
+	// Fields that identify the client:
+	Version minecraft = 1;
+	Version forge = 2;
+	Version eln = 3;
+	OS os = 4;
+	string uuid = 5;  // May be empty.
+	// The message(s) that we're trying to log.
+	// Any of these fields may be empty.
+	CrashReport crash_report = 6;
+	Analytics analytics = 7;
+}
+
+// This is a straight textual copy of Forge's crash report, for later perusal.
+message CrashReport {
+	string report = 1;
+}
+
+// There should be a config section to disable/limit this.
+message Analytics {
+	// Included mainly for stability/crash analysis.
+	message OtherMod {
+		string name = 1;
+		Version version = 2;
+	}
+	repeated OtherMod mods = 1;
+	// Included for performance analysis.
+	message Circuit {
+		int32 matrix_size = 1;
+		int32 avg_time_usec = 2;
```

Comment:
Average time is not nearly as useful I don't think. In the least, average and maximum, but I would prefer to split it between matrix time, thevenin and norton process time, and other tick time. We don't currently have power sources nor intersystem, so we can't quite see where this will come into play, or if it will be relevant in this section or elsewhere in a protobuff - but I would value the input nonetheless to know how those times compare with different circuits.

Also, if I recall correctly, profiling the game while running in debug versions caused a pretty big performance hit, so we may want to be careful how we collect this data. Some benchmarks and testing in the mna code may be advised before making that change.

Reply:
Well, it doesn't have to be profiled all the time. Statistical profilers can be made to have arbitrarily low performance impact.

I've split it out as suggested.

Reply:
True, having the profiler running 24/7 is probably where the performance hit was. Just something to consider, I guess.